src/traits.jl

```diff
@@ -109,83 +109,149 @@ to_top(::Type{Colorant{T,N}}) where {T,N} = Colorant{T,N}
 to_top(c::Colorant) = to_top(typeof(c))
 
 # eltype(RGB{Float32}) -> Float32
-eltype(::Type{Colorant{T}}) where {T          }   = T
-eltype(::Type{Colorant{T,N}}) where {T,N        } = T
+eltype(::Type{Colorant{T}}) where {T}       = T
+eltype(::Type{Colorant{T,N}}) where {T,N}   = T
 @pure eltype(::Type{C}) where {C<:Colorant} = eltype(supertype(C))
 
 eltype(c::Colorant) = eltype(typeof(c))
 
-# eltypes_supported(Colorant{T<:X}) -> T<:X (pre 0.6) or X (post 0.6)
+# eltypes_supported(Colorant{T<:X}) -> X
 @pure eltypes_supported(::Type{C}) where {C<:Colorant} =
     Base.parameter_upper_bound(base_colorant_type(C), 1)
+eltypes_supported(::Type{RGB24})   = N0f8
+eltypes_supported(::Type{Gray24})  = N0f8
+eltypes_supported(::Type{ARGB32})  = N0f8
+eltypes_supported(::Type{AGray32}) = N0f8
 
 eltypes_supported(c::Colorant) = eltypes_supported(typeof(c))
 
-@pure issupported(::Type{C}, ::Type{T}) where {C<:Colorant,T} = T <: eltypes_supported(C)
+"""
+    issuported(C::Type, T::Type)::Bool
```

Comment:
```suggestion
    issupported(C::Type, T::Type)::Bool
```

Reply:
It might be good idea to aggregate these eltype related functions around [here](https://github.com/JuliaGraphics/ColorTypes.jl/blob/5b2e50d62b8fd47a860c2e0543446144d8627c87/src/types.jl#L535-L549).

`eltype_ub` and `eltypes_supported` are different. I am concerned about using the reflection-like technique, despite the fact that we definitely know the supported eltypes. That also concerns #152.